### PR TITLE
FAI-251: Score Cards: Setup showcase for PMML editor

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-251

@ederign @paulovmr This adds the "showcase" type webapp to the `pmml-editor` module.

I first had a `pmml-editor-showcase` module but it seemed an overkill; so I moved it all to the `pmml-editor` module itself but exclude the webapp related stuff when building for "production". A new `build:showcase` script bundles the necessary stuff together to build a webapp I can use with `webpack-dev-server`.

The `PMMLEditor` itself shows a `TextArea` with a string in it.. wow!

It is temporary ;-)

### CODE REVIEW

File: packages/envelope-bus/webpack.config.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = merge(common, {
     "channel/index": "./src/channel/index.ts",
     "envelope/index": "./src/envelope/index.ts",
     "api/index": "./src/api/index.ts",
+    "common/index": "./src/common/index.ts",
```
Comment (manstis) [inline packages/envelope-bus/webpack.config.js:27]: I'm not a fan of exposing `common` but I need access to `EnvelopeBusMessageManager` when [stubbing](https://github.com/kiegroup/kogito-tooling/pull/267/files#diff-b47cbd8fe48189b4e8d5f54ce9b10fe4R23) the PMML editor for use in a "showcase" type webapp...... other ideas welcome (IDK, an example of mocking `KogitoEditorChannelApi` would work equally well!)

File: packages/pmml-editor/package.json [hunk 2]
```diff
@@ -27,7 +27,9 @@
     "test:clearCache": "jest --clearCache",
     "build:fast": "rm -rf dist && webpack",
     "build": "yarn run lint && yarn test && yarn run build:fast",
-    "build:prod": "yarn run build --mode production --devtool none"
+    "build:prod": "yarn run build --mode production --devtool none",
+    "build:showcase": "yarn run build --mode development",
```
Comment (manstis) [inline packages/pmml-editor/package.json:31]: I add a new build to the `pmml-editor` to generate a webapp to use to develop the editor.. (like our beloved _showcases_)

File: packages/pmml-editor/src/showcase/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import {PMMLEditor} from "../editor";
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {EnvelopeBusMessageManager} from "@kogito-tooling/envelope-bus/dist/common";
+
+const manager: EnvelopeBusMessageManager<KogitoEditorChannelApi, KogitoEditorChannelApi> = new EnvelopeBusMessageManager((msg: any) => console.log(msg));
+
+let editor: PMMLEditor;
+
+ReactDOM.render(
```
Comment (manstis) [inline packages/pmml-editor/src/showcase/index.tsx:30]: Ta da! The showcase webapp.

File: packages/pmml-editor/webpack.config.js [hunk 4]
```diff
@@ -14,17 +14,64 @@
...
+            globalObject: "this"
+        },
+        module: {
+            rules: [...pfWebpackOptions.patternflyRules]
+        }
+    });
+
+module.exports = (env, argv) => {
+
+    if (argv.mode === 'development') {
```
Comment (manstis) [inline packages/pmml-editor/webpack.config.js]: I only bundle the webapp stuff when building for `development` (see `build:showcase` [here](https://github.com/kiegroup/kogito-tooling/pull/267/files#diff-09f0ee0148822b7fd19f8d758b782452R31)).

File: packages/pmml-editor/webpack.config.js [hunk 5]
```diff
@@ -14,17 +14,64 @@
...
+                        path.join(__dirname, "./dist/showcase/static")
+                    ],
+                    compress: true,
+                    port: 9001,
+                    open: true
+                }
+            });
+    }
+
+    if (argv.mode === 'production') {
```
Comment (manstis) [inline packages/pmml-editor/webpack.config.js]: Otherwise, for `production`, only a bundle for the editor itself is created.

**Review by paulovmr — APPROVED** [review]
LGTM

File: packages/pmml-editor/webpack.config.js [hunk 6]
```diff
@@ -14,17 +14,57 @@
...
+          { from: "./src/showcase/static/index.html", to: "./showcase/index.html" },
+          { from: "./src/showcase/static/favicon.ico", to: "./showcase/favicon.ico" }
+        ])
+      ],
+      devtool: "source-map",
+      devServer: {
+        historyApiFallback: false,
+        disableHostCheck: true,
+        watchContentBase: true,
+        contentBase: [path.join(__dirname, "./dist/showcase"), path.join(__dirname, "./dist/showcase/static")],
```
Comment (kelvah) [inline packages/pmml-editor/webpack.config.js]: I don't think you have a `static` folder there, you are copying its content inside `showcase`.

**Review by kelvah — APPROVED** [review]
It looks fine to me. Just a minor observation below.

File: packages/pmml-editor/webpack.config.js (hunk 6 above)
Comment (manstis) [reply packages/pmml-editor/webpack.config.js]: Good spot. Corrected.

**Pull request merged**